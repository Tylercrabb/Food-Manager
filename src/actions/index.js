 import {API_KEY} from '../config'
 
export const FETCH_FRIDGE_SUCCESS = 'FETCH_FRIDGE_SUCCESS';

export const fetchFridgeSuccess = items => ({
    type: FETCH_FRIDGE_SUCCESS,
    items
});

export const fetchFridgeInventory = () => (dispatch, getState) => {
    const authToken = getState().auth.authToken;
    dispatch(loading());
    return fetch(`https://fridgeapp-backend.herokuapp.com/api/item`,{
        method: 'GET',
        headers: {
            // Provide our auth token as credentials
            Authorization: `Bearer ${authToken}`
        }
    })

        .then(res => {
            if (!res.ok) {
                return Promise.reject(res.statusText);
            }
            return res.json();
        })
        .then(items => {
            dispatch(fetchFridgeSuccess(items));
            dispatch(stopLoading())
        });
    }

export const FETCH_PANTRY_SUCCESS = 'FETCH_PANTRY_SUCCESS';

export const fetchPantrySuccess = items => ({
    type: FETCH_PANTRY_SUCCESS,
    items
});

export const fetchPantryInventory = () => (dispatch, getState) => {
    const authToken = getState().auth.authToken;
    dispatch(loading())
    return fetch(`https://fridgeapp-backend.herokuapp.com/api/pantry`, {
        method: 'GET',
        headers: {
            // Provide our auth token as credentials
            Authorization: `Bearer ${authToken}`
        }
    })
        .then(res => {
            if (!res.ok) {
                return Promise.reject(res.statusText);
            }
        return res.json()
        })
        .then(items => {
            dispatch(fetchPantrySuccess(items));
            dispatch(stopLoading())
        });
    }

export const DELETE_PANTRY_ITEM_SUCCESS = 'DELETE_PANTRY_ITEM_SUCCESS';

export const deletePantryItemSuccess = items => ({
    type: DELETE_PANTRY_ITEM_SUCCESS,
    items
});

    export const deletePantryItem = (item) => (dispatch, getState) => {
        const authToken = getState().auth.authToken;
        
        return fetch(`https://fridgeapp-backend.herokuapp.com/api/pantry/${item.id}`, {
            method: 'DELETE',
            headers: {
                // Provide our auth token as credentials
                Authorization: `Bearer ${authToken}`
            }
        })
            .then(res => {
                if (!res.ok) {
                    return Promise.reject(res.statusText);
                }
                return res
            })
            .then( ()=> {
                dispatch(deletePantryItemSuccess(item.id));
            })
        }
        

export const DELETE_FRIDGE_ITEM_SUCCESS = 'DELETE_FRIDGE_ITEM_SUCCESS';

export const deleteFridgeItemSuccess = items => ({
    type: DELETE_FRIDGE_ITEM_SUCCESS,
    items
});

    export const deleteFridgeItem = (item) => (dispatch, getState) => {
        const authToken = getState().auth.authToken;
        
       return fetch(`https://fridgeapp-backend.herokuapp.com/api/item/${item.id}`, {
            method: 'DELETE',
            headers: {
                // Provide our auth token as credentials
                Authorization: `Bearer ${authToken}`
            }
        })
            .then(res => {
                if (!res.ok) {
                    return Promise.reject(res.statusText);
                }
                
            })
            .then( ()=> {
                dispatch(deleteFridgeItemSuccess(item.id));
            })
        }

export const GET_RECIPES_SUCCESS = 'GET_RECIPES_SUCCESS';

export const getRecipeSuccess = results => ({
    type: GET_RECIPES_SUCCESS,
    results
});

    export const getRecipes = () => (dispatch, getState) => {
        function getNames(arr){
            let results = [];
            for(let i = 0; i < arr.length; i++){
                results.push(arr[i].itemName)
            }return results
        }
        let fridgeInvNames = getNames(getState().food.fridgeInventory) 
        let pantryInvNames = getNames(getState().food.pantryInventory) 
        let totalInv = fridgeInvNames.concat(pantryInvNames);
        let query = totalInv.join('%2C');
        dispatch(loading())
       return fetch(`https://spoonacular-recipe-food-nutrition-v1.p.rapidapi.com/recipes/findByIngredients?number=5&ranking=1&ingredients=${query}`, {
            method: 'GET',
            
            headers: {
                'X-RapidAPI-Key': API_KEY
            }
        })
            .then(res => {
                if (!res.ok) {
                    return Promise.reject(res.statusText);
                }
                return res.json()
            })
            .then( results => {
                if(results.length < 1){
                    dispatch(setErrorMessage('Oh no! No recipes found using your inventory!'))
                }
                dispatch(getRecipeSuccess(results))
                dispatch(stopLoading())
            })
        }


export const CLEAR_RECIPES ='CLEAR_RECIPES'

export const clearRecipes = () => ({
    type: CLEAR_RECIPES
})


export const SET_ERROR_MESSAGE = 'SET_ERROR_MESSAGE';

export const setErrorMessage = errorMessage => ({
    type: SET_ERROR_MESSAGE,
    errorMessage
});

export const CLEAR_ERROR_MESSAGE = 'CLEAR_ERROR_MESSAGE';

export const clearErrorMessage = errorMessage => ({
    type: CLEAR_ERROR_MESSAGE,
    errorMessage
});

export const LOADING = 'LOADING';

export const loading = flip => ({
    type: LOADING,
    flip
});

export const STOP_LOADING = 'STOP_LOADING';

export const stopLoading = flip => ({
    type: STOP_LOADING,
    flip
});

export const SET_ADDING = 'SET_ADDING';

export const setAdding = addTo => ({
    type: SET_ADDING,
    addTo
})
            
export const SET_EXPIRING_ITEMS = 'SET_EXPIRING_ITEMS';

export const setExpiringItems = items => ({
    type: SET_EXPIRING_ITEMS,
    items
})

export const CLEAR_EXPIRING_ITEMS = 'CLEAR_EXPIRING_ITEMS';

export const clearExpiringItems = items => ({
    type: CLEAR_EXPIRING_ITEMS,
    items
})
           