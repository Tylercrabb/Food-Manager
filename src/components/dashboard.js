import React from 'react';
import {connect} from 'react-redux';
import { Route, Link} from 'react-router-dom';
import {loading, fetchFridgeInventory, fetchPantryInventory} from '../actions/index'
import {clearErrorMessage, clearExpiringItems} from '../actions'
// components
import AddForm from './addForm';
import requiresLogin from './requires-login'
import RecipeViewer from './recipeViewer';
import FridgeInventory from './FridgeInventory'
import PantryInventory from './PantryInventory'
import ExpirationItems from './expiration'


// icons
import fridge from '../images/fridge.png';
import pantry from '../images/pantry.png';
import plus from '../images/plus.png'
import garbageCan from '../images/garbagecan.png'
import oven from '../images/oven.png'
import grocery from '../images/grocery.jpg'



export class Dashboard extends React.Component {
  componentWillMount(){
    this.props.dispatch(clearErrorMessage())
    this.props.dispatch(fetchFridgeInventory())
    this.props.dispatch(fetchPantryInventory())
  }


  render() {
    
    return (
      
      <section className="Dashboard">
          <nav className="App-header">
            <Link className="Fridge-logo" to="/fridge"><img src={fridge}  alt="logo" /></Link>
            <Link className="Pantry-logo" to ="/pantry"><img src={pantry}  alt="logo" /></Link>
            <Link className="Recipe-logo" to="/recipe"><img src={oven} alt ="oven"/></Link>
            <Link className="Garbagecan-logo"to="/dashboard"><img src={garbageCan} alt ='garbage'/></Link>
            <Link className="Add-logo" to="/add"><img src={plus} alt ='add'/></Link>
          </nav>
      <main>
        <Route  path='/fridge' component={FridgeInventory}/>
        <Route  path='/pantry' component={PantryInventory}/>
        <Route  path ='/add' component={AddForm} />
        <Route  path ='/recipe' component={RecipeViewer} />
        <Route exact path ='/' component = {FridgeInventory} />
        <Route exact path ='/dashboard' component = {ExpirationItems} />
      </main>
      </section>
    );
  }
}

const mapStateToProps = state => {
  const {currentUser} = state.auth;
  return {
      username: state.auth.currentUser.username,
      name: `${currentUser.firstName} ${currentUser.lastName}`,
      isLoading: state.food.isLoading
      
  };
};

export default requiresLogin()(connect(mapStateToProps)(Dashboard))
