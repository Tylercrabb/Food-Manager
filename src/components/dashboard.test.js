import React from 'react';
import {shallow, mount} from 'enzyme';
import {loading, fetchFridgeInventory, fetchPantryInventory} from '../actions/index'
import {clearErrorMessage, clearExpiringItems} from '../actions'
import Enzyme from 'enzyme';
import {Dashboard} from './dashboard';
import './setupTests'






describe('<Dashboard />', () => {
    const dispatch = jest.fn();

    it('Renders without crashing', () => {
        shallow(<Dashboard dispatch= {dispatch}/>);
    });

    // it('the page correctly renders all links and routes', () => {
    //     const wrapper = shallow(<Dashboard dispatch={dispatch}/>);
    //     expect(wrapper).toMatchSnapshot()
    // });

    it('calls the life cycle methods when mounted', () => {
        const testLifeCycle = jest.fn();
        const wrapper = shallow(<Dashboard dispatch = {testLifeCycle}/>);
        expect(testLifeCycle.mock.calls.length).toBe(3);
    })

});



// expect(wrapper.find('').length).toEqual(1)