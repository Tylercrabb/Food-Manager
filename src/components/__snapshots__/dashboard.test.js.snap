// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> the page correctly renders all links and routes`] = `
<section
  className="Dashboard"
>
  <nav
    className="App-header"
  >
    <Link
      className="Fridge-logo"
      replace={false}
      to="/fridge"
    >
      <img
        alt="logo"
        src="fridge.png"
      />
    </Link>
    <Link
      className="Pantry-logo"
      replace={false}
      to="/pantry"
    >
      <img
        alt="logo"
        src="pantry.png"
      />
    </Link>
    <Link
      className="Recipe-logo"
      replace={false}
      to="/recipe"
    >
      <img
        alt="oven"
        src="oven.png"
      />
    </Link>
    <Link
      className="Garbagecan-logo"
      replace={false}
      to="/dashboard"
    >
      <img
        alt="garbage"
        src="garbagecan.png"
      />
    </Link>
    <Link
      className="Add-logo"
      replace={false}
      to="/add"
    >
      <img
        alt="add"
        src="plus.png"
      />
    </Link>
  </nav>
  <main>
    <Route
      component="FridgeInventory"
      path="/fridge"
    />
    <Route
      component={[Function]}
      path="/pantry"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "WrappedComponent": [Function],
          "defaultProps": Object {
            "destroyOnUnmount": true,
            "enableReinitialize": false,
            "forceUnregisterOnUnmount": false,
            "form": "add",
            "getFormState": [Function],
            "keepDirtyOnReinitialize": false,
            "onSubmitFail": [Function],
            "persistentSubmitErrors": false,
            "pure": true,
            "shouldAsyncValidate": [Function],
            "shouldError": [Function],
            "shouldValidate": [Function],
            "shouldWarn": [Function],
            "touchOnBlur": true,
            "touchOnChange": false,
            "updateUnregisteredFields": false,
          },
          "displayName": "ReduxForm",
          "render": [Function],
          Symbol(Symbol.hasInstance): [Function],
        }
      }
      path="/add"
    />
    <Route
      component={[Function]}
      path="/recipe"
    />
    <Route
      component="FridgeInventory"
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/dashboard"
    />
  </main>
</section>
`;

exports[`<Dashboard /> the page correctly renders all links and routes 1`] = `
<section
  className="Dashboard"
>
  <nav
    className="App-header"
  >
    <Link
      className="Fridge-logo"
      replace={false}
      to="/fridge"
    >
      <img
        alt="logo"
        src="fridge.png"
      />
    </Link>
    <Link
      className="Pantry-logo"
      replace={false}
      to="/pantry"
    >
      <img
        alt="logo"
        src="pantry.png"
      />
    </Link>
    <Link
      className="Recipe-logo"
      replace={false}
      to="/recipe"
    >
      <img
        alt="oven"
        src="oven.png"
      />
    </Link>
    <Link
      className="Garbagecan-logo"
      replace={false}
      to="/dashboard"
    >
      <img
        alt="garbage"
        src="garbagecan.png"
      />
    </Link>
    <Link
      className="Add-logo"
      replace={false}
      to="/add"
    >
      <img
        alt="add"
        src="plus.png"
      />
    </Link>
  </nav>
  <main>
    <Route
      component={[Function]}
      path="/fridge"
    />
    <Route
      component={[Function]}
      path="/pantry"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "WrappedComponent": [Function],
          "defaultProps": Object {
            "destroyOnUnmount": true,
            "enableReinitialize": false,
            "forceUnregisterOnUnmount": false,
            "form": "add",
            "getFormState": [Function],
            "keepDirtyOnReinitialize": false,
            "onSubmitFail": [Function],
            "persistentSubmitErrors": false,
            "pure": true,
            "shouldAsyncValidate": [Function],
            "shouldError": [Function],
            "shouldValidate": [Function],
            "shouldWarn": [Function],
            "touchOnBlur": true,
            "touchOnChange": false,
            "updateUnregisteredFields": false,
          },
          "displayName": "ReduxForm",
          "render": [Function],
          Symbol(Symbol.hasInstance): [Function],
        }
      }
      path="/add"
    />
    <Route
      component={[Function]}
      path="/recipe"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/dashboard"
    />
  </main>
</section>
`;
